lib/analysis.js

```diff
@@ -52,6 +53,7 @@ function AnalysisFactory(DatabaseServiceClass) {
 
 AnalysisFactory.prototype.create = function(configuration, definition, callback) {
     debug('Using configuration=%j', configuration);
+    var logger = createLogger(configuration.logger).child({ analysis: definition.type, user: configuration.user });
```

Comment:
For now, it's OK to log each individual analysis separately. Have in mind that given an analysis id you can retrieve its dependencies and its dependent analyses.